compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -616,24 +616,14 @@ declare_lint! {
     ///
     /// ### Explanation
     ///
-    /// It was expected that the marked code would emit a lint. This expectation
-    /// has not been fulfilled.
+    /// The `#[expect]` attribute can be used to create a lint expectation. The
+    /// expectation is fulfilled, if a `#[warn]` attribute at the same location
+    /// would result in a lint emission. If the expectation is unfulfilled,
+    /// because no lint was emitted, this lint will be emitted on the attribute.
```

Comment:
I think this needs to be rephrased. Isn't the expectation only fulfilled if the lint isn't allowed? Like

```rust
#![allow(some_lint)]

#[expect(some_lint)]
fn foo() {
    something_that_would_emit_some_lint();
}
```

will actually warn, because the `expect` is not fulfilled. The sentence as written signals that the expectation would be fulfilled and this would compile warning free

Reply:
This question was one of the central questions, discussed in https://github.com/rust-lang/rust/issues/115980. The decision was that expect should act like a lint level. A strong motivator for this was [use case 4](https://hackmd.io/@xFrednet/expect-attr-use-cases#Use-case-4-Suppress-lints-from-CI). The short summary is, that if `#[expect]` is based on the outer lint level, it can be annoying to use with command line arguments, that might only enable some lints in the CI.

I think this model is also easier to explain. If you slap a `#[warn]` attribute there, you would expect a waning, done. You don't need to consider the outer scope.

Reply:
gotcha, so the impl already reflects that, just the docs were outdated?

Reply:
Correct. The previous docs actually wanted to express the same thing, but at that time the model wasn't as clear to me, which is why the docs were a bit vague.